realm/src/main/java/io/realm/DynamicRealmObject.java

```diff
@@ -34,31 +30,60 @@
  */
 public class DynamicRealmObject extends RealmObject {
 
-     Realm realm;
-     Row row;
-
     /**
      * Creates a dynamic Realm object based on a existing object.
      *
      * @param obj Realm object to convert to a dynamic object. Only objects managed by Realm can be used.
-     * @throws IllegalArgumentException if object isn't managed by a Realm.
+     * @throws IllegalArgumentException if object isn't managed by Realm.
      */
     public DynamicRealmObject(RealmObject obj) {
         if (obj == null) {
             throw new IllegalArgumentException("Non-null object must be provided.");
         }
-        Row row = RealmObject.getRow(obj);
+        Row row = obj.row;
         if (row == null) {
-            throw new IllegalArgumentException("A object managed by Realm must be provided. This is a standalone object.");
+            throw new IllegalArgumentException("A object managed by Realm must be provided. This " +
+                    "is a standalone object.");
         }
-        this.realm = RealmObject.getRealm(obj);
-        this.row = (row instanceof CheckedRow) ? (CheckedRow) row : ((UncheckedRow) row).convertToChecked();
+        this.realm = obj.realm;
+        this.row = (row instanceof CheckedRow) ? row : ((UncheckedRow) row).convertToChecked();
     }
 
     // Create a dynamic object. Only used internally
-    DynamicRealmObject(Realm realm, CheckedRow row) {
+    DynamicRealmObject() {
+
+    }
+
+    DynamicRealmObject(BaseRealm realm, Row row) {
         this.realm = realm;
-        this.row = row;
+        this.row = (row instanceof CheckedRow) ? (CheckedRow) row : ((UncheckedRow) row).convertToChecked();
+    }
+
+    /**
+     * Returns the value for the given field. To avoid casting the value use one of the typed
+     * getters instead.
+     *
+     * @param fieldName Name of the field.
+     * @return The field value. Private types will be converted to their boxed variants.
+     */
+    public Object get(String fieldName) {
```

Comment:
`public <E> E get(String fieldName)` is a valid method declaration in Java and in most cases, E is decided by type inference.

```
Long value = realmObject.get("foo");
```

if `get(String)` returns for example `String` object, simply `ClassCastException` will be thrown.

This is not an overloaded return type because actual return type is `Object`.